portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +25,17 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages = new ArrayList<>(Arrays.asList("one", "two", "three"));
```

Comment:
```suggestion
  private List<String> messages = new ArrayList<>(Arrays.asList("one", "two", "three"));
```

This is not Google style guide, but a common best practice. If declaring the variable type to be a more generic interface, it is more flexible. The same as your method parameters. Imagine there is a method that takes LinkedList, then this variable can not be passed into this method. But if both variable and method parameter are declared as List, then they can fit each other.

Reply:
I will be sure to do this in the future--made the change.